File: src/main/kotlin/project/jvmstatic/JvmStatic.java
package project.jvmstatic;

public class JvmStatic {
    public static void main(String[] args) {
        System.out.println(HiObject.INSTANCE.hi());
        System.out.println(HelloClass.Companion.hello());

        System.out.println(HiObject.hi());
        System.out.println(HelloClass.hello());
    }
}

File: src/main/kotlin/project/jvmstatic/JvmStatic.kt
package project.jvmstatic

import kotlin.jvm.JvmStatic

object HiObject {
    @JvmStatic
    fun hi() = "hi!"
}

class HelloClass {
    companion object {
        @JvmStatic
        fun hello() = "hello!"
    }
}

fun main() {
    println(HiObject.hi())
    println(HelloClass.hello())
}